### PR TITLE
Add health check API

### LINKED ISSUE #38: Add a health check API
Currently, there exists no health check API for load-watcher. This can be used for example to have liveness/readiness checks in Kubernetes. This can be an additional API in the existing web server.

### PR DESCRIPTION
**Pull request by zorro786**
Fixes #38 
Also updates go version in go.mod and CI for tests.

### CODE REVIEW

**Review by wangchen615 — APPROVED** [review]
Thanks so much for adding the health check APIs

File: pkg/watcher/internal/metricsprovider/signalfx.go [hunk 1]
```diff
@@ -191,6 +191,28 @@ func (s signalFxClient) FetchAllHostsMetrics(window *watcher.Window) (map[string
...
+		return -1, err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return -1, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return -1, fmt.Errorf("received response code: %v", resp.StatusCode)
+	}
+	resp.Body.Close()
```
Comment (lenhattan86) [inline pkg/watcher/internal/metricsprovider/signalfx.go:212]: should we use defer and bring it up to 206?

File: pkg/watcher/metricsprovider.go [hunk 2]
```diff
@@ -50,11 +50,14 @@ type MetricsProviderClient interface {
 	FetchHostMetrics(host string, window *Window) ([]Metric, error)
 	// Fetch metrics for all hosts
 	FetchAllHostsMetrics(window *Window) (map[string][]Metric, error)
+	// Get metric provider server health status
+	// Returns 0 if healthy, -1 if unhealthy along with error if any
+	Health() (int, error)
```
Comment (lenhattan86) [inline pkg/watcher/metricsprovider.go:55]: if it is 0 or 1? why don't we just get rid of it?
error can bit nil or not nil?

File: pkg/watcher/watcher.go [hunk 3]
```diff
@@ -297,6 +300,17 @@ func (w *Watcher) handler(resp http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Simple server status handler
+func (w *Watcher) healthCheckHandler(resp http.ResponseWriter, r *http.Request) {
+	status, err := w.client.Health()
+	if status != 0 {
```
Comment (lenhattan86) [inline pkg/watcher/watcher.go:306]: we can combine line 305 & 306 to a single line.

**Review by lenhattan86 — CHANGES_REQUESTED** [review]

File: pkg/watcher/internal/metricsprovider/prometheus.go [hunk 4]
```diff
@@ -133,6 +134,22 @@ func (s promClient) FetchAllHostsMetrics(window *watcher.Window) (map[string][]w
...
+func (s promClient) Health() (int, error) {
+	req, err := http.NewRequest("HEAD", DefaultPromAddress, nil)
+	if err != nil {
+		return -1, err
+	}
+	resp, _, err := s.client.Do(context.Background(), req)
+	if err != nil {
+		return -1, err
+	}
+	defer resp.Body.Close()
```
[the remark this answers is not in the document]
Comment (zorro786) [reply pkg/watcher/internal/metricsprovider/prometheus.go:146]: Not possible, resp is nil if err != nil

File: pkg/watcher/internal/metricsprovider/signalfx.go (hunk 1 above)
Comment (zorro786) [reply pkg/watcher/internal/metricsprovider/signalfx.go:212]: This is actually redundant, we are not reading response to close it. I will remove this line.

File: pkg/watcher/metricsprovider.go (hunk 2 above)
Comment (zorro786) [reply pkg/watcher/metricsprovider.go:55]: error will help to customize message internally. also they may not be present always if unhealthy . Status codes will be used for metrics in the future.

File: pkg/watcher/internal/metricsprovider/prometheus.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (zorro786) [reply pkg/watcher/internal/metricsprovider/prometheus.go:146]: Anyway, this line is not needed as we are not reading body.

[the remark this answers is not in the document]
Comment (zorro786) [reply pkg/watcher/internal/metricsprovider/prometheus.go]: This is actually not needed, as there is no response expected.

**Comment by lenhattan86** [comment]
looks good to me.

**Review by lenhattan86 — APPROVED** [review]

**Pull request merged**